rfcs/convergence/authoring-stories.md

```diff
@@ -261,17 +261,112 @@ export const parameters = { controls: { expanded: true } };
 
 ### 5. how to author e2e suites for those stories
 
-TBA
+> - This section assumes that the storybook stories forms the core documentation of Fluent converged.
+> This may change in the future so the measures proposed in this section might no longer be relevant
+> if this assumption is no longer true.
 
-### 6. how to properly annotate stories with TS metadata to get the best DX possible
+`react-menu` already has entire test suites dedicated to testing against storybook. Since each PR will build and
+deploy a storybook (storybook build for `react-components` package), the e2e suite (implemented with cypress) for converged components will execute against current PR deployed storybook instance. This
+includes the constraint that until a component is added to `react-components` it cannot be tested in CI.
 
-TBA
+The e2e testing solution for `react-menu` can be used for any converged component using storybook, and is also used by
+`react-accordion` and `react-popover`. The purposes of E2E tests currently is to run interaction tests against the browser.
+Visual regression tests are handled by screener in `app-vrtests` and is out of the scope of this RFC.
+
+We propose to write an NX workspace generator to setup the e2e testing folder structure and dependencies for each package. The process is quite simple currently but still needs to be done manually.
+
+How to write E2E tests is out of the scope of the RFC. Testing is encouraged to use publicly viewable stories as much as possible. However there will be edge cases and scenarios that will be tested which require stories that are accessible to the E2E test runner but not viewable by public consumers. We provide a proposal to handle this in a later section in this RFC.
+
+### 6. dissecting big story files into smaller ones
+
+For easier maintainability, each story should be in its own file. Common utilities can be shared in their own modules.
```

Comment:
Alright, so your proposal is that we keep one single story file for each component ?

Reply:
Probably. I think it’s best to have one story file, but we could still import story functions from other files (that wouldn’t have the `.stories.tsx` extension). I need like one more week to try it out and formulate a proposal. Do you see any downsides to this approach right now?

Reply:
As you say `Button` has many stories, however in the `Menu` case there are equally as many stories but the stories are longer and more complicated and soe stories need the use of components outside the story export itself. 

`Menu.stories.tsx` contains 11 stories in a file that is 400 lines long, we can keep it, but it might continue to grow especially is we want to test stories that are not part of the public documentation